src/solvers/dgmulti/flux_differencing.jl

```diff
@@ -190,29 +223,17 @@ end
 # and jth reference coordinate, respectively. These are geometric terms which
 # appear when using the chain rule to compute physical derivatives as a linear
 # combination of reference derivatives.
-@inline function get_contravariant_vector(element, orientation, mesh::DGMultiMesh{1})
-  @unpack rxJ = mesh.md
-  return 2 * SVector(rxJ[1, element]) # the 1D contravariant vector reduces to a scaling.
+@inline function get_contravariant_vector(element, orientation, mesh::DGMultiMesh{NDIMS}) where {NDIMS}
+  # note that rstxyzJ = [rxJ, sxJ, txJ; ryJ syJ tyJ; rzJ szJ tzJ], so that this will return
+  # 2 * SVector{2}(rxJ[1, element], ryJ[1, element]) in 2D.
+  @unpack rstxyzJ = mesh.md
+  return 2 * SVector{NDIMS}(getindex.(rstxyzJ[:, orientation], 1, element))
```

Comment:
This allocates, doesn't it? How does the performance summary look like? You could use some `ntuple` if you want to avoid duplicating similar code for every dimension (as we do for `get_node_vars` etc.).

Reply:
It does allocate, but I think it's because of the `2 *` in front. `rstxzyJ` is a `SMatrix` so `getindex` and broadcasting seem to be fast. 

I'll try to move the `2 *` around to avoid this.

Reply:
I tried `return 2 * SVector{NDIMS}(ntuple(ref_coord_index -> rstxyzJ[ref_coord_index, orientation][1, element], NDIMS))` instead. This also doesn't allocate, but seems to take a little longer when benchmarking `Trixi.get_contravariant_vector` (around 2.5 ns vs 3.378 ns for the broadcast). 

Do you have a preference?